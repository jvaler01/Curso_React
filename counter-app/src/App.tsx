import PropTypes from 'prop-types';
import React from "react";

export const App: React.FunctionComponent<AppProps> = ({ title, subTitle, name }) => {

    // console.log(props);

    return (
        <>
            <h1 data-testid="test-title"> { title } </h1>
            {/* <code>{ JSON.stringify( newMessage ) }</code> */}
            <p>{ subTitle }</p>
            <p>{ subTitle }</p>
            <p>{ name }</p>
        </>
    )
}
App.propTypes = {
    title: PropTypes.string.isRequired,
    subTitle: PropTypes.string,
}
App.defaultProps = {
    name: 'Reac App',
    subTitle: 'No hay subtítulo',
    // title: 'No hay título',
}
type AppProps = {
    title: string;
    subTitle?: string;
    name: string;
};