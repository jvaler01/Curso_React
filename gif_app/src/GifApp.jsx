import {useState} from "react";
import {AddCategory} from "./components/AddCategory.jsx";

export const GifApp = () => {
    const [categories, setCategories] = useState([ 'Star Wars' ]);
    const onAddCategory = ( newCategory ) => {
        setCategories([...categories, newCategory])
    }

    return (
        <>
            <h1>GifApp</h1>
            <AddCategory onNewCategory={ onAddCategory } />
            <ol>
                {
                    categories.map(item => {
                        return <li key={ item }> { item }</li>
                    })
                }
            </ol>
        </>
    )
}