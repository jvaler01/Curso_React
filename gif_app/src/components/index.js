export * from './AddCategory.jsx';
export * from './GifItem.jsx';
export * from './GifGrid.jsx';