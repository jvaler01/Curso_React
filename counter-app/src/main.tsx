import React from "react";
import ReactDOM from "react-dom/client";
import { CounterApp } from './CounterApp';
import {App} from "./App.tsx";
import './styles.css';

ReactDOM.createRoot(document.getElementById('root')!).render(
    <React.StrictMode>
        <CounterApp value={ 20 } />
        <App title="Hola" name="Reac App"></App>
    </React.StrictMode>
);