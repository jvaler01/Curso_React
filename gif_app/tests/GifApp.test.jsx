import { render } from '@testing-library/react';
import { GifApp } from '../src/GifApp';

describe('Tests in GifExpertApp', () => {
    test('should', () => {
        const { container } = render( <GifApp /> );
        expect(container).toMatchSnapshot();
    });
});