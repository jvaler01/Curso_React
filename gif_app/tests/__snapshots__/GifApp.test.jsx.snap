// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests in GifExpertApp should 1`] = `
<div>
  <h1>
    GifApp
  </h1>
  <form
    aria-label="form"
  >
    <input
      placeholder="Buscar gifs"
      type="text"
      value=""
    />
  </form>
  <h3>
    Star Wars
  </h3>
  <h2>
    Cargando...
  </h2>
  <div
    class="card-grid"
  />
</div>
`;
